media-video/mpv/mpv-0.34.1_p20221105.ebuild

```
@@ -0,0 +1,288 @@
+		if use test; then
+			einfo "Skipping -DNDEBUG due to USE=test"
+		else
+			append-cppflags -DNDEBUG # treated specially
+		fi
+	fi
+
+	if use raspberry-pi; then
+		append-cflags -I"${ESYSROOT}"/opt/vc/include
+		append-ldflags -L"${ESYSROOT}"/opt/vc/lib
+	fi
+
+	mpv_feature_multi() {
+		local use set
+		for use in ${1} ${2}; do
+			use ${use} || set=disabled
+		done
+		echo -D${3-${2}}=${set-enabled}
+	}
+
+	local emesonargs=(
+		$(meson_use cli cplayer)
+		$(meson_use libmpv)
+		$(meson_use test tests)
+
+		$(meson_feature cli html-build)
+		$(meson_feature cli manpage-build)
+		-Dpdf-build=disabled
+
+		-Dbuild-date=false
+
+		# misc options
+		$(meson_feature archive libarchive)
+		$(meson_feature bluray libbluray)
+		$(meson_feature cdda)
+		-Dcplugins=enabled
+		$(meson_feature dvb dvbin)
+		$(meson_feature dvd dvdnav)
+		$(meson_feature gamepad sdl2-gamepad)
+		$(meson_feature iconv)
+		$(meson_feature javascript)
+		-Dlibavdevice=enabled
+		$(meson_feature lcms lcms2)
+		-Dlua=$(usex lua "${ELUA}" disabled)
+		$(meson_feature rubberband)
+		-Dsdl2=$(use gamepad || use sdl && echo enabled || echo disabled) #857156
+		$(meson_feature uchardet)
+		-Dvapoursynth=disabled # only available in overlays
+		$(meson_feature vector)
+		$(meson_feature zimg)
+		$(meson_feature zlib)
+
+		# audio output
+		$(meson_feature alsa)
+		$(meson_feature coreaudio)
+		$(meson_feature jack)
+		$(meson_feature openal)
+		$(meson_feature pipewire)
+		$(meson_feature pulseaudio pulse)
+		$(meson_feature sdl sdl2-audio)
+		$(meson_feature sndio)
+
+		# video output
+		$(meson_feature X x11)
+		$(meson_feature aqua cocoa)
+		$(meson_feature drm)
+		$(meson_feature drm gbm)
+		$(meson_feature jpeg)
+		$(meson_feature libcaca caca)
+		$(meson_feature libplacebo)
+		$(meson_feature mmal rpi-mmal)
+		$(meson_feature sdl sdl2-video)
+		-Dsixel=disabled # TODO? needs keywording/testing
+		$(meson_feature wayland)
+		$(meson_feature xv)
+
+		-Dgl=$(use egl || use libmpv || use opengl || use raspberry-pi &&
+			echo enabled || echo disabled)
+		$(meson_feature egl)
+		$(mpv_feature_multi egl X egl-x11)
+		$(mpv_feature_multi egl drm egl-drm)
+		$(mpv_feature_multi egl wayland egl-wayland)
+		$(meson_feature libmpv plain-gl)
+		$(mpv_feature_multi opengl X gl-x11)
+		$(mpv_feature_multi opengl aqua gl-cocoa)
+		$(meson_feature raspberry-pi rpi)
+
+		$(meson_feature vulkan)
+		$(meson_feature vulkan shaderc)
+
+		# hardware decoding
+		$(meson_feature nvenc cuda-hwaccel)
+		$(meson_feature nvenc cuda-interop)
+
+		$(meson_feature vaapi)
+		$(mpv_feature_multi vaapi X vaapi-x11)
+		$(mpv_feature_multi 'vaapi X' egl vaapi-x-egl)
+		$(mpv_feature_multi 'vaapi egl' drm vaapi-drm)
+		$(mpv_feature_multi 'vaapi egl' wayland vaapi-wayland)
+
+		$(meson_feature vdpau)
+		$(mpv_feature_multi vdpau opengl vdpau-gl-x11)
+
+		$(mpv_feature_multi aqua opengl videotoolbox-gl)
+	)
+
+	meson_src_configure
+}
+
+src_test() {
+	edo "${BUILD_DIR}"/mpv --no-config -v --unittest=all-simple
```

I should've done this before but maybe:
```suggestion
	# Not mentioned in many places, see https://github.com/mpv-player/mpv/blob/master/DOCS/man/options.rst#debugging
	edo "${BUILD_DIR}"/mpv --no-config -v --unittest=all-simple
```

I usually like to do it so we can look in future in case they add some more detail or change the recommended invocation for packagers.

---

Yeah I haven't really looked much into this bit, so good to keep the link in mind.